addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -539,21 +539,66 @@ private HueDevice itemToDevice(Item item) {
     }
 
     /**
-     * Gets all items that match our tag
+     * Gets and syncs all items tagged for voice.
      *
      * @return
      */
-    private Collection<Item> getTaggedItems() {
-        Collection<Item> items = new LinkedList<Item>();
+    private synchronized TreeMap<Integer, HueDevice> getHueDevices() {
+        TreeMap<Integer, HueDevice> returnMap = new TreeMap<Integer, HueDevice>();
+        HashMap<String, Item> taggedItems = new HashMap<String, Item>();
+
+        // if we modify our internal map, persist it to disk
+        boolean modified = false;
+
+        // get all tagged items
         for (Item item : itemRegistry.getItems()) {
             for (String tag : item.getTags()) {
                 if (ArrayUtils.contains(SUPPORTED_TAGS, tag)) {
-                    items.add(item);
+                    taggedItems.put(item.getName(), item);
+                    if (!deviceMap.containsValue(item.getName())) {
+                        // hue devices are assigned a numeric number starting with 1, if a device is
+                        // removed that number is not used again. Not sure how high this id can get
+                        // not worrying about it here
+                        Integer next = deviceMap.size() == 0 ? new Integer(1)
```

Comment:
Java has the concept of auto boxing for this, the new integer(1) is not needed and you can also just add 1 to the lastKey without any conversions in your code.